joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -85,28 +86,46 @@ trait AstForStatementsCreator(implicit withSchemaValidation: ValidationMode) { t
       .getOrElse(Ast(retNode))
   }
 
-  private def astForCatchClause(catchClause: BabelNodeInfo): Ast =
-    astForNodeWithFunctionReference(catchClause.json("body"))
+  private def astForCatchClause(catchClause: BabelNodeInfo): Ast = {
+    val blockNode = createBlockNode(catchClause)
+    scope.pushNewBlockScope(blockNode)
+    localAstParentStack.push(blockNode)
+    val paramAst = safeObj(catchClause.json, "param")
+      .map { param => astForNodeWithFunctionReference(Obj(param)) }
+      .getOrElse(Ast())
+    val bodyAsts           = createBlockStatementAsts(catchClause.json("body")("body"))
+    val blockStatementAsts = paramAst +: bodyAsts
+    setArgumentIndices(blockStatementAsts)
+    localAstParentStack.pop()
+    scope.popScope()
+    blockAst(blockNode, blockStatementAsts)
+  }
 
   protected def astForTryStatement(tryStmt: BabelNodeInfo): Ast = {
     val tryNode = createControlStructureNode(tryStmt, ControlStructureTypes.TRY)
     val bodyAst = astForNodeWithFunctionReference(tryStmt.json("block"))
+    setOrderExplicitly(bodyAst, 1)
```

Comment:
Why do we still need to set t he order explicitly here and for `catchAst` and `finalAst`?

Reply:
Right, not required anymore. But I need to set the index. Will fix.